### PR TITLE
[[ Bug 23193 ]] Add support for App Tracking Transparency

### PR DESCRIPTION
**Pull request by livecodepanos**
This patch adds support for App Tracking Transparency. If the 'NSUserTrackingUsageDescription' key is present in the plist, the app shows an authorization dialog, asking the user to allow the app to track them.

### CODE REVIEW

**Comment by livecodepanos** [comment]
Needs https://github.com/livecode/livecode-ide/pull/2153

File: docs/dictionary/function/iphoneTrackingAuthorizationStatus.lcdoc [hunk 1]
```diff
@@ -0,0 +1,46 @@
+Name: iphoneTrackingAuthorizationStatus
+
+Type: function
+
+Syntax: iphoneTrackingAuthorizationStatus()
+
+Summary:
+Returns the current tracking authorization status of the calling
+aplication.
```
Comment (livecodeian) [inline docs/dictionary/function/iphoneTrackingAuthorizationStatus.lcdoc]: 'application'

**Comment by livecodeian** [comment]
@livecode-vulcan review ok a7d3109

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeian ok a7d310966354ef19b95a6035abe38aacfe70f87b

**Comment by livecode-vulcan** [comment]
😞 test failure a7d310966354ef19b95a6035abe38aacfe70f87b

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/1001)
* try-community-armv7-android-ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/409)
* try-community-arm64-android-ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/408)
* try-community-x86-android-ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/407)
* try-community-x86_64-android-ndk16r15: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/408)
* try-community-js-emscripten-sdk1.35: [pending](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2526)
* try-community-universal-ios-iphoneos14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.5/builds/4)
* try-community-universal-ios-iphoneos14.4: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.4/builds/55)
* try-community-universal-ios-iphonesimulator14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.5/builds/4)
* try-community-universal-ios-iphonesimulator14.4: [pending](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.4/builds/55)
* try-community-universal-mac-macosx10.9: [failure](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2553)
* try-community-x86-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2542)
* try-community-x86_64-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2540)
* try-community-x86-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2577)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2513)

**Comment by livecodeian** [comment]
@livecode-vulcan review ok  3f9e51f

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeian ok 3f9e51fa9f57398cb51418f42eb11e1a5113a287

**Comment by livecode-vulcan** [comment]
😞 test failure 3f9e51fa9f57398cb51418f42eb11e1a5113a287

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/1011)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/419)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/418)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/417)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/418)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2536)
* try-community-universal-ios-iphoneos14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.5/builds/14)
* try-community-universal-ios-iphoneos14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.4/builds/65)
* try-community-universal-ios-iphonesimulator14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.5/builds/14)
* try-community-universal-ios-iphonesimulator14.4: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.4/builds/65)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2563)
* try-community-x86-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2552)
* try-community-x86_64-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2550)
* try-community-x86-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2587)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2523)

**Comment by livecodepanos** [comment]
@livecode-vulcan retry 3f9e51f

**Comment by livecode-vulcan** [comment]
😞 test failure 3f9e51fa9f57398cb51418f42eb11e1a5113a287

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/1012)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/420)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/419)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/418)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/419)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2537)
* try-community-universal-ios-iphoneos14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.5/builds/15)
* try-community-universal-ios-iphoneos14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.4/builds/66)
* try-community-universal-ios-iphonesimulator14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.5/builds/15)
* try-community-universal-ios-iphonesimulator14.4: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.4/builds/66)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2564)
* try-community-x86-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2553)
* try-community-x86_64-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2551)
* try-community-x86-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2588)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2524)

**Comment by livecodepanos** [comment]
@livecode-vulcan retry 3f9e51f

**Comment by livecode-vulcan** [comment]
😞 test failure 3f9e51fa9f57398cb51418f42eb11e1a5113a287

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/1014)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/422)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/421)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/420)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/421)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2539)
* try-community-universal-ios-iphoneos14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.5/builds/17)
* try-community-universal-ios-iphoneos14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.4/builds/68)
* try-community-universal-ios-iphonesimulator14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.5/builds/17)
* try-community-universal-ios-iphonesimulator14.4: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.4/builds/68)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2566)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2555)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2553)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2590)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2526)

**Comment by livecodepanos** [comment]
@livecode-vulcan retry 3f9e51f

**Comment by livecode-vulcan** [comment]
😞 test failure 3f9e51fa9f57398cb51418f42eb11e1a5113a287

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/1016)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/424)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/423)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/422)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/423)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2541)
* try-community-universal-ios-iphoneos14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.5/builds/19)
* try-community-universal-ios-iphoneos14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.4/builds/70)
* try-community-universal-ios-iphonesimulator14.5: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.5/builds/19)
* try-community-universal-ios-iphonesimulator14.4: [pending](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.4/builds/70)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2568)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2557)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2555)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2592)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2528)

**Comment by livecodeian** [comment]
@livecode-vulcan review ok 0183bbc

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeian ok 0183bbc73d7810b450d5fb4c40b2ec130c5d8902

**Comment by livecode-vulcan** [comment]
😞 test failure 0183bbc73d7810b450d5fb4c40b2ec130c5d8902

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/1017)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/425)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/424)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/423)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/424)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2542)
* try-community-universal-ios-iphoneos14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.5/builds/20)
* try-community-universal-ios-iphoneos14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.4/builds/71)
* try-community-universal-ios-iphonesimulator14.5: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.5/builds/20)
* try-community-universal-ios-iphonesimulator14.4: [pending](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.4/builds/71)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2569)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2558)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2556)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2593)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2529)

**Comment by livecodeian** [comment]
@livecode-vulcan review ok 8aa50c1

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeian ok 8aa50c14474e163e364b98020c74d1d216c7aab6

**Comment by livecode-vulcan** [comment]
😎 test success 8aa50c14474e163e364b98020c74d1d216c7aab6

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/1018)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/426)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/425)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/424)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/425)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2543)
* try-community-universal-ios-iphoneos14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.5/builds/21)
* try-community-universal-ios-iphoneos14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.4/builds/72)
* try-community-universal-ios-iphonesimulator14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.5/builds/21)
* try-community-universal-ios-iphonesimulator14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.4/builds/72)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2570)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2559)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2557)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2594)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2530)

**Pull request merged**